src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ProcessedEnvelopeNotificationHandler.java

```diff
@@ -32,6 +33,7 @@
     private final EnvelopeFinaliserService envelopeFinaliserService;
     private final ObjectMapper objectMapper;
     private final MessageAutoCompletor messageCompletor;
+    private ExecutorService executor = Executors.newFixedThreadPool(4);
```

Comment:
perhaps `static final` ;)

Reply:
p.s. people where concerned about having more than 1 thread and iirc there is a ticket to investigate best/better approach for that. not the issue here but worth mentioning so we are all aware

Reply:
I'll start with one thread then

Reply:
done